src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -37,42 +37,69 @@ public class EditCommand extends Command {
-            + "[" + PREFIX_NAME + "NAME] "
-            + "[" + PREFIX_PHONE + "PHONE] "
-            + "[" + PREFIX_EMAIL + "EMAIL] "
-            + "[" + PREFIX_ADDRESS + "ADDRESS] "
-            + "[" + PREFIX_TAG + "TAG]"
-            + "[" + PREFIX_GRADE + "TEST_NAME: GRADE]...\n"
-            + "Example: " + COMMAND_WORD + " 1 "
-            + PREFIX_PHONE + "91234567 "
-            + PREFIX_EMAIL + "johndoe@example.com "
-            + PREFIX_GRADE + "EoY: 80";
+            + "[" + OPTION_PRINT_NAME + "] or "
+            + "[" + OPTION_PRINT_PHONE + "] or "
+            + "[" + OPTION_PRINT_EMAIL + "] or "
+            + "[" + OPTION_PRINT_TAG + "] or "
+            + "[" + OPTION_PRINT_GRADE + "] or "
+            + "[" + OPTION_PRINT_ADDRESS + "]\n"
+            + "Example: " + COMMAND_WORD + " 1 " + OPTION_PRINT_EMAIL;
 
     public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Person: %1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_EDITED_BUT_MORE_THAN_ONE = "Please ensure that only one field is edited at most.";
+
     public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
 
     private final Index index;
     private final EditPersonDescriptor editPersonDescriptor;
 
+    private final boolean isPrintNameRequested;
+    private final boolean isPrintPhoneRequested;
+    private final boolean isPrintEmailRequested;
+    private final boolean isPrintAddressRequested;
+    private final boolean isPrintTagRequested;
+    private final boolean isPrintGradeRequested;
+
     /**
      * @param index of the person in the filtered person list to edit
      * @param editPersonDescriptor details to edit the person with
      */
-    public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
+    public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor, boolean isPrintNameRequested,
+            boolean isPrintPhoneRequested, boolean isPrintEmailRequested, boolean isPrintAddressRequested,
+                       boolean isPrintTagRequested, boolean isPrintGradeRequested) {
         requireNonNull(index);
         requireNonNull(editPersonDescriptor);
 
         this.index = index;
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+        this.isPrintNameRequested = isPrintNameRequested;
+        this.isPrintPhoneRequested = isPrintPhoneRequested;
+        this.isPrintEmailRequested = isPrintEmailRequested;
+        this.isPrintAddressRequested = isPrintAddressRequested;
+        this.isPrintTagRequested = isPrintTagRequested;
+        this.isPrintGradeRequested = isPrintGradeRequested;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
+        if (isPrintNameRequested) {
+            return new CommandResult(getPersonName(model, index));
```

Comment:
Maybe this might be clearer for a user to use
```suggestion
            return new CommandResult("Edit name: "getPersonName(model, index));
```